### PR TITLE
Passed `Omnibox`'s timeout through to `ResultPreviewsManager`

### PR DESCRIPTION
**Pull request by btaillon**
Made sure that `ResultPreviewsManager` has the same timeout for previews as `Omnibox`.

https://coveord.atlassian.net/browse/JSUI-2729

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/magicbox/SuggestionsManagerTest.ts [hunk 1]
```diff
@@ -465,171 +456,209 @@ export function SuggestionsManagerTest() {
...
+            it('times out if the previews takes too long to resolve', async done => {
+              mouseFocusSuggestion(0);
+              await deferAsync();
+              jasmine.clock().tick(displayAfterDuration);
+              await waitForPreviewsPopulateEvent();
+              jasmine.clock().tick(timeout);
+              await waitForSelectionUpdated();
+              expect($$(env.root).findClass(previewClassName).length).toEqual(0);
+              done();
+            });
```
Comment (btaillon) [inline unitTests/magicbox/SuggestionsManagerTest.ts]: This is the only added test here. No test was removed or modified.

File: unitTests/magicbox/SuggestionsManagerTest.ts [hunk 2]
```diff
@@ -404,17 +406,6 @@ export function SuggestionsManagerTest() {
...
-          let previewsBySuggestion: ISearchResultPreview[][];
-          const textPreviews = [['4K TV', 'Set of 4 pairs of socks'], ['The right spoon', '🔶', 'Right angle SATA cable']];
-          function buildPreviews() {
-            previewsBySuggestion = textPreviews.map(previewTextsForSuggestion => previewTextsForSuggestion.map(buildPreview));
-          }
-
-          const previewsPromisesWaitTimes = [1, 2];
-          function createPreviewsPromise(suggestionId: number) {
-            return Utils.resolveAfter(previewsPromisesWaitTimes[suggestionId], previewsBySuggestion[suggestionId]);
-          }
```
Comment (btaillon) [inline unitTests/magicbox/SuggestionsManagerTest.ts]: This, and several other pieces of code, wasn't removed. I moved every tests in `describe('with previews')` into a new describe block under it named `describe('within timeout delay')`. I moved some functions and beforeEach calls out of `within timeout delay` into `with previews` so they could work with the new `describe('timing out')`.

tl;dr:
Went from
- **with previews**
  - ...functions
  - `beforeEach`
  - `afterEach`
  - ...tests
To
- **with previews**
  - ...most old functions
  - `beforeEach`
  - `afterEach`
  - **timing out**
    - ...new functions
    - `beforeEach`
    - ...new tests
  - **within timeout delay**
    - ...a few old functions
    - `beforeEach`
    - ...old tests
